syncstorage/storage/sql/__init__.py

```diff
@@ -593,55 +670,91 @@ def delete_item(self, session, userid, collection, item):
     #
     # Administrative/maintenance methods.
     #
+    def purge_expired_items(self, grace_period=0, max_per_loop=1000):
+        """Purges items with an expired TTL from the bso and batch_upload_items
+           table(s).
 
-    @with_session
-    def purge_expired_items(self, session, grace_period=0, max_per_loop=1000):
-        """Purges items with an expired TTL from the database."""
-        # Get the set of all BSO tables in the database.
-        # This will be different depending on whether sharding is done.
-        if not self.dbconnector.shard:
-            tables = set(("bso",))
-        else:
-            tables = set(self.dbconnector.get_bso_table(i).name
-                         for i in xrange(self.dbconnector.shardsize))
-            assert len(tables) == self.dbconnector.shardsize
-        # Purge each table in turn, summing rowcounts.
-        # We set an upper limit on the number of iterations, to avoid
-        # getting stuck indefinitely on a single table.
-        total_affected = 0
+           The grace period for batch_upload_items is hard coded to 1.5 time
+           the current upper limit(2 hours) for a batch session's lifetime.
```

Comment:
nit: space after "limit"?

Does this imply that the batch session lifetime of 2 hours is also hard-coded, implicity at least?  Seems like something we should lift into config if so.